slatedb-dst/src/dst.rs

```diff
@@ -272,12 +272,11 @@ impl DefaultDstDistribution {
     /// for more information.
     fn sample_write(&self, state: &SizedBTreeMap<Vec<u8>, Vec<u8>>) -> DstAction {
         let mut write_ops = Vec::new();
-        let write_option = self.get_write_options();
-        let put_probability = 0.8;
+        let write_options = self.get_write_options();
         let mut remaining_bytes =
             self.sample_log10_uniform(1..self.options.max_write_batch_bytes) as i64;
         while remaining_bytes > 0 {
-            let is_put = self.rand.rng().random_bool(put_probability);
+            let is_put = self.rand.rng().random_bool(self.get_put_probability());
```

Comment:
I think you need to move the probability to a separate line since we're using refmut and the Rand bool already borrows on this line.